R/init.R

```diff
@@ -19,20 +19,34 @@
 #' @param rhino_version When using an existing `renv.lock` file,
 #' Rhino will install itself using `renv::install(rhino_version)`.
 #' You can provide this argument to use a specific version / source, e.g.`"Appsilon/rhino@v0.4.0"`.
+#' @param force Create a `rhino` app without checking for if working directory is home directory.
+#' Defaults to `FALSE` to prevent creating a `rhino` app in the home directory
+#' Set to `TRUE` to create a `rhino` app in the home directory.
 #' @return None. This function is called for side effects.
 #'
 #' @export
 init <- function(
   dir = ".",
   github_actions_ci = TRUE,
-  rhino_version = "rhino"
-) {
-  init_impl(
-    dir = dir,
-    github_actions_ci = github_actions_ci,
-    rhino_version = rhino_version,
-    new_project_wizard = FALSE
-  )
+  rhino_version = "rhino",
+  force = FALSE
+)  {
+  is_home <- is_dir_home(dir = dir)
+
+  if (!is_home || force) {
+    init_impl(
+      dir = dir,
+      github_actions_ci = github_actions_ci,
+      rhino_version = rhino_version,
+      new_project_wizard = FALSE
+    )
+  } else {
+    cli::cli_abort(
+      "Refusing to create {.pkg rhino} in {.path {dir}}!
+      You are in your home directory. Please set {.code force = TRUE} if you
+      want to create a {.pkg rhino} app in your home directory."
+    )
```

Comment:
"Your are in your home directory" suggests that it's your current working directory; it will be somewhat inaccurate when calling `rhino::init("~")` from somewhere else.
```suggestion
    cli::cli_abort(c(
      "Refusing to create a Rhino app in home directory {.path {dir}}!",
      i = "Set {.code force = TRUE} to force initialization."
    ), call = FALSE)
```